Adding pause and optimistic concurrency on content update code

sturufous opened the pull request:
Hope my somewhat parrot-style implementation of the pause functionality in the transcription service was done correctly.

Fosol commented on services/net/transcription/TranscriptionManager.cs:
```diff
@@ -76,12 +91,24 @@ public override async Task RunAsync()
                     }
+                    else if (topics.Length == 0)
+                    {
+                        this.Consumer.Stop();
```
Make sure this doesn't cause issues if there are no topics configured.

sturufous replied:
Not sure how I would remedy this.

Fosol replied:
If there are no topics, then it'll attempt to stop the consumer.  You can test this and see what happens.  If no error is thrown, then you don't have to do anything.  If an error is thrown then we'll need to figure out a way to handle this.

sturufous commented on services/net/transcription/TranscriptionManager.cs:
```diff
@@ -20,13 +20,21 @@ namespace TNO.Services.Transcription;
+    /// <summary>
+    /// get - Lookup values from the API.
+    /// </summary>
+    public string[] Topics { get; private set; } = Array.Empty<string>();
```
I only put this in because it was used in the fragment on line 74 - 76. As that is not required I'm removing both.

sturufous commented on services/net/transcription/TranscriptionManager.cs:
```diff
@@ -76,12 +91,24 @@ public override async Task RunAsync()
                         var tps = String.Join(',', topics);
```
Just an artifact copied over from an older version of the command service. Removed it.

sturufous commented on services/net/transcription/TranscriptionManager.cs:
```diff
@@ -60,9 +68,16 @@ public override async Task RunAsync()
+                // An API request or failures have requested the service to stop.
+                this.Logger.LogInformation("The service is stopping: '{Status}'", this.State.Status);
+                this.State.Stop();
+                this.Topics = Array.Empty<string>();
```
Removed.

sturufous commented:
@Fosol I pushed an update. I tested with no topics and nothing bad happened. I just got a message saying the consumer had stopped. I did have to change the way the topics were split out from the topic string. As _options.Topics defaults to the empty string the following line:

var topics = _options.Topics.Split(',');

returns an array with one element in it containing "". I changed this to:

var topics = _options.Topics.Split(',', StringSplitOptions.RemoveEmptyEntries);

so that removing the line:

Service__TranscriptionTopic=transcription

from the .env file results in an empty array after spitting _options.Topics.
